### DISCUSSION TITLE
bind() - how should we determine if the given address is valid?

### DISCUSSION THREAD

**Discussion by livelykitten** [ANLAB-KAIST/KENSv3#163] [Project Q&A 2. Transport Layer (TCP)] [answered]
My understanding of bind() is that
1. sockets can be bound to all network interface of the host, by giving INADDR_ANY as an address to bind
2. sockets can be bound to one specific IP address of the host, and it must be given in an argument
3.  any IP address that does not belong to the host results in an error

How can you know if the given IP address is one of the host's and is valid?
I have read the tip below, but you don't have destination IP address given in bind()..
https://github.com/ANLAB-KAIST/KENSv3/wiki/Tip:-Figuring-out-the-source-IP-address-to-use

**Comment by Pusnow** [comment, marked as answer]
You can use `getIPAddr()` to retrieve host's IP addresses. You can get all IP addresses by using port from 0 to the number whose return value is error.

**Reply by livelykitten** [reply]
Thank you!! I would have spent a lot of time to figure it out myself
